services/orchestrator/src/main/java/org/eclipse/osc/orchestrator/OrchestratorPlugin.java

```diff
@@ -0,0 +1,39 @@
+package org.eclipse.osc.orchestrator;
+
+import org.eclipse.osc.services.ocl.loader.Ocl;
+
+/**
+ * This interface describes orchestrator plugin in charge of interacting with backend fundamental APIs.
+ */
+public interface OrchestratorPlugin {
+
+    /**
+     * Register a managed service using the provided OCL descriptor.
+     *
+     * @param ocl the OCL model describing the managed service.
+     * @return the service ID on the CSP infrastructure.
+     */
+    String registerManagedService(Ocl ocl);
```

Comment:
does this include preparing infrastructure and deploying the manged service? Seems we didn't talk about deploying before.

Reply:
The idea behind this method is to be "global": this method is responsible of:
1. creating the infra as described in OCL
2. download the artifacts from the locations in OCL `artifacts`
3. bootstrapping the resources needed to actually start the service

NB: the Orchestrator (and generally OSC runtime) is supposed to run "in the CloudServiceProvider" infra. End user doesn't know OSC.

Reply:
Maybe we can introduce Packer here to help build image on the CSP with the artifact and boot VM with that image. As far as I know, Spinnaker Rosco use this way.
https://www.packer.io/
https://github.com/spinnaker/rosco

Reply:
Good idea. I think it could make sense in orchestrator plugin level.

Reply:
I propose to create later issues and PRs about first `OrchestratorPlugins` we can create.